module/x/peggy/keeper/cosmos-originated.go

```diff
@@ -43,6 +44,12 @@ func (k Keeper) DenomToERC20Lookup(ctx sdk.Context, denom string) (bool, string,
 	tc1, err := types.PeggyDenomToERC20(denom)
 
 	if err != nil {
+
+		// If denom is inj, return INJ Contract address from Params.
+		if denom == "inj" {
```

Comment:
Remove INJ hardcoding. Introduce a parameter that sets `CosmosCoinDenom` in the keeper. Check denom with this parameter and only then takes action.